contrib/offline/manage-offline-files.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+CURRENT_DIR=$(cd $(dirname $0); pwd)
+OFFLINE_FILES_DIR="${CURRENT_DIR}/offline-files"
+FILES_LIST="${CURRENT_DIR}/temp/files.list"
+PATHS_LIST="${CURRENT_DIR}/temp/paths.list"
+NGINX_PORT=8080
+
+# step 1 : generate files.list and paths.list by generate_list.sh
+./generate_list.sh
+
+# step 2 : download files
+rm -rf ${OFFLINE_FILES_DIR}
+mkdir ${OFFLINE_FILES_DIR}
+wget -P ${OFFLINE_FILES_DIR} -i ${FILES_LIST}
+
+# step 3:  move files to the specified path of offline files
+while read -r line; do
+    FILE_NAME="$(echo ${line} | awk -F"/" '{print $NF}')"
+    FILE_PATH="${OFFLINE_FILES_DIR}/$(echo ${line} | sed s@"${FILE_NAME}"@@)"
+    mkdir -p ${FILE_PATH}
+    mv "${OFFLINE_FILES_DIR}/${FILE_NAME}" ${FILE_PATH}
+done <<< "$(cat ${PATHS_LIST})"
+
+# step 4 : run nginx container server
+if command -v nerdctl 1>/dev/null 2>&1; then
+    runtime="nerdctl"
+elif command -v podman 1>/dev/null 2>&1; then
+    runtime="podman"
```

Comment:
Detecting the runtime is actually a good approach since we are not usually in control of the environment (ansible machine) on which kubespray is running.